### PR TITLE
Retry on ETIMEDOUT as well

### PR DESCRIPTION
**Pull request by kerneltime**

### CODE REVIEW

**Comment by msterin** [comment]
I still do not understand why and how this timeout can happen. Let's nor rush with the patches, and instead check if this is indeed VMCI failure - with latest kernels all the tests should pass since the bug is not there, right ?

**Comment by kerneltime** [comment]
Yup or an older kernel.

File: vmdk_plugin/vmdkops/esx_vmdkcmd.go [hunk 1]
```diff
@@ -107,14 +107,14 @@ func (vmdkCmd EsxVmdkCmd) Run(cmd string, name string, opts map[string]string) (
...
-         if errno == syscall.ECONNRESET {
-            msg += fmt.Sprintf(" retryCount=%d Retrying Request\n", retryCount)
-            log.Warnf(msg)
-            time.Sleep(3 * time.Millisecond)
-         } else {
-            log.Errorf(msg)
-            return nil, errors.New(msg)
-         }
+			if errno == syscall.ECONNRESET || errno == syscall.ETIMEDOUT {
+				msg += fmt.Sprintf(" retryCount=%d Retrying Request\n", retryCount)
```
Comment (govint) [inline vmdk_plugin/vmdkops/esx_vmdkcmd.go]: nit - define a function to validate if given error is among errors identified as retryable.

Comment (govint) [inline vmdk_plugin/vmdkops/esx_vmdkcmd.go]: Pls. print what error was returned in the log below.

File: vmdk_plugin/vmdkops/esx_vmdkcmd.go [hunk 2]
```diff
@@ -107,14 +107,14 @@ func (vmdkCmd EsxVmdkCmd) Run(cmd string, name string, opts map[string]string) (
...
+				log.Warnf(msg)
+				time.Sleep(3 * time.Millisecond)
+			} else {
+				log.Errorf(msg)
+				return nil, errors.New(msg)
+			}
 		}
```
Comment (govint) [inline vmdk_plugin/vmdkops/esx_vmdkcmd.go]: Add a warning log that retries are "done and giving up"?

**Pull request closed without merging**